modules/browsemap/src/main/java/com/sygic/modules/browsemap/viewmodel/BrowseMapFragmentViewModel.kt

```diff
@@ -57,31 +58,30 @@ class BrowseMapFragmentViewModel internal constructor(
     val positionLockFabEnabled: MutableLiveData<Boolean> = MutableLiveData()
     val zoomControlsEnabled: MutableLiveData<Boolean> = MutableLiveData()
 
+    var onMapClickListener: OnMapClickListener? = null
+    var detailsViewFactory: DetailsViewFactory? = null
+
     val poiDataObservable: LiveData<PoiData> = SingleLiveEvent()
 
     val dialogFragmentListener: DialogFragmentListener = object : DialogFragmentListener {
         override fun onDismiss() {
-            extendedMapDataModel.removeOnClickMapMarker()
+            mapDataModel.removeOnClickMapMarker()
         }
     }
 
-    private var onMapClickListener: OnMapClickListener? = null
-    private var detailsViewFactory: DetailsViewFactory? = null
     private var poiDetailsView: UiObject? = null
 
     init {
-        attributesTypedArray?.let {
-            mapSelectionMode =
-                    it.getInt(R.styleable.BrowseMapFragment_sygic_map_selectionMode, MapSelectionMode.MARKERS_ONLY)
-            positionOnMapEnabled = it.getBoolean(R.styleable.BrowseMapFragment_sygic_positionOnMap_enabled, false)
-
-            compassEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compass_enabled, false)
-            compassHideIfNorthUp.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_compass_hideIfNorthUp, false)
-            positionLockFabEnabled.value =
-                    it.getBoolean(R.styleable.BrowseMapFragment_sygic_positionLockFab_enabled, false)
-            zoomControlsEnabled.value = it.getBoolean(R.styleable.BrowseMapFragment_sygic_zoomControls_enabled, false)
-            it.recycle()
-        }
+        initComponent.resolveAttributes(app)
+        mapSelectionMode = initComponent.mapSelectionMode
+        positionOnMapEnabled = initComponent.positionOnMapEnabled
+        compassEnabled.value = initComponent.compassEnabled
+        compassHideIfNorthUp.value = initComponent.compassHideIfNorthUp
+        positionLockFabEnabled.value = initComponent.positionLockFabEnabled
+        zoomControlsEnabled.value = initComponent.zoomControlsEnabled
+        onMapClickListener = initComponent.onMapClickListener
+        detailsViewFactory = initComponent.detailsViewFactory
+        initComponent.recycle()
```

Comment:
realne to recycle ale nevrati ten object do nejakeho poolu ze? mozno by sa neskor aj nieco take mohlo urobit ak to budeme chciet naozaj recyklovat

Reply:
žádný pool zatím neexistuje, defakto to jen "vyresetuje" obsah toho komponentu, to je vše. můžeme tu metodu nazvat jinak, např. destroy pokud se ti to tu hodí více?